src/commands/delete.ts

```diff
@@ -53,26 +54,28 @@ function getItemToDelete(contextItem: TektonNode, selectedItems: TektonNode[]):
 
 async function doDelete(items: TektonNode[], toRefresh: Refreshable, commandId?: string): Promise<void | string> {
   if (items) {
-    const taskRunList = await getTaskRunResourceList();
     let message: string;
-    let refResource = true;
+    let hasAnyReference = false;
+    let pipelineList: TknPipeline[];
+    const hasAnyTaskOrClusterTask = items.some(item => item.contextValue === ContextType.TASK || item.contextValue === ContextType.CLUSTERTASK);
+    if (hasAnyTaskOrClusterTask) {
+      pipelineList = await getPipelineList();
+    }
     const toDelete = new Map<TektonNode, CliCommand>();
     for (const item of items) {
       const deleteCommand = getDeleteCommand(item);
       if (deleteCommand) {
         toDelete.set(item, deleteCommand);
       }
-      if (refResource && checkRefResource()) {
-        if (referenceOfTaskAndClusterTaskInCluster(item, taskRunList)) {
-          refResource = false;
-        }
+      if (!hasAnyReference && checkRefResource() && pipelineList?.length !== 0 && (item.contextValue === ContextType.TASK || item.contextValue === ContextType.CLUSTERTASK)) {
```

Comment:
Wouldn't it be better and simpler if you initialize the `pipelineList` as an empty array so you avoid using the `?` safe operator?